linera-core/src/client.rs

```diff
@@ -434,6 +434,24 @@ where
                             error!("Fail to synchronize new block after notification");
                             return false;
                         }
+                        match local_node
+                            .storage_client()
+                            .await
+                            .read_certificate(hash)
```

Comment:
Even downloading from local storage feels a bit expensive. Can we add a TODO and create an issue to optimize this?

Reply:
Done: https://github.com/linera-io/linera-protocol/issues/940